src/open_inwoner/js/components/search/filter-mobile.js

```diff
@@ -0,0 +1,35 @@
+export class FilterMobile {
+  static selector = '.filter__mobile--toggle'
+
+  constructor(node) {
+    this.node = node
+    this.filterList = node.closest('.filter') // Find the closest parent with class 'filter'
+    this.isSmallscreen = window.innerWidth < 768 // Check if it's a mobile screen
+    this.node.addEventListener('click', this.toggleOpen.bind(this))
+    document.addEventListener('keydown', this.filterClosing.bind(this), false)
+  }
+
+  toggleOpen(event) {
+    if (this.isSmallscreen) {
+      event.preventDefault()
+      console.log('this is mobile')
+      this.filterList.classList.toggle('filter--open')
+    }
+  }
+
+  filterClosing(event) {
+    if (
+      this.isSmallscreen &&
+      event.type === 'keydown' &&
+      event.key === 'Escape'
+    ) {
+      this.filterList.classList.remove('filter--open')
+    }
+  }
+}
+
+/**
+ * Controls the toggling of filter lists on mobile to view more
+ */
+const filterToggles = document.querySelectorAll(FilterMobile.selector)
+;[...filterToggles].forEach((filterToggle) => new FilterMobile(filterToggle))
```

Comment:
Let's start removing this cargo-culted code pattern and just use forEach on the nodeList:

````js
document.querySelectorAll(FilterMobile.selector).forEach((filterToggle) => new FilterMobile(filterToggle))
````

Later we'll move them to the index.

This goes for every `querySelectorAll()` & `;[...xxx]` wart.

Reply:
@Bartvaderkin Okay, that would mean that, in future, we would need a bunch of imported Classes + initialized constants (with DOMContentLoaded event listener?) in the index file?

Reply:
Yes, in the future we'll add them all to the list in `open_inwoner/js/components/index.js` so we have a good overview of all the components, and they activate on page load as well as in htmx fragements.